addons/website/models/ir_model.py

```diff
@@ -30,6 +30,10 @@ def get_base_url(self):
             return self._get_http_domain() or super().get_base_url()
         if 'website_id' in self and self.website_id.domain:
             return self.website_id._get_http_domain()
+        # Check if we are working from the frontend, we do not want a fallback
+        current_website = self.env['website'].get_current_website(fallback=False)
+        if current_website and current_website.domain:
+            return current_website._get_http_domain()
```

Comment:
I'd go with:
```suggestion
        # Before returning the ICP, use the related website if we are in the frontend
        website = ir_http.get_request_website()
        if website and website.domain:
            return website._get_http_domain()
```

`ir_http.get_request_website()` is the way to be sure we are in the frontend (technically, if request has a website, it means we went through the frontend dispatcher)

Can you see if it fits your need? If not, it probably means that you actually are not in the frontend but just rely on the context, which is not what the comment is hinting

Reply:
Alright makes sense. Since the get_request_website is a local function in the ir_http file, should I import it with `from odoo.addons.website.models.ir_http import get_request_website` or `from .ir_http import get_request_website`? I don't know the Odoo conventions for local imports

Reply:
![image](https://github.com/odoo/odoo/assets/30048408/43bc6ef1-5d41-4ddc-80f6-578e574036cb)
You can just check other imports of this method in the codebase and do the same, which should follow what the docstring say

Otherwise, if there were not this particularity, you could go with any of `from . import ir_http` / `from odoo.addons.website.models import ir_http`